test/metabase/test.clj

```diff
@@ -136,7 +136,8 @@
  [qp
   compile
   preprocess
-  process-query]
+  process-query
+  query->expected-cols]
```

Comment:
I'm curious, why aren't you using `qp.test-util/card-with-source-metadata-for-query`? I think the point of that function is to prepare card's `result_metadata`.

Reply:
no reasons tbh, I just didn't know about that function.

looks like a neat one. Will replace to use it.

Reply:
Changed https://github.com/metabase/metabase/pull/27343/commits/d8766d67ed428a5b0dc56776774e737e58f88daa and https://github.com/metabase/metabase/pull/27343/commits/1e40f7fc3ae564ad14278ff583b638761e992467